pkg/bundle/release.go

```diff
@@ -6,64 +6,61 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 
 	semver "github.com/blang/semver/v4"
 	"github.com/google/uuid"
 	"github.com/openshift/oc/pkg/cli/admin/release"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
 
 	"github.com/RedHatGov/bundle/pkg/cli"
 	"github.com/RedHatGov/bundle/pkg/config"
 	"github.com/RedHatGov/bundle/pkg/config/v1alpha1"
 	"github.com/RedHatGov/bundle/pkg/image"
 )
 
-// This file is for managing OCP release related tasks
-
-// import(
-//   "github.com/openshift/cluster-version-operator/pkg/cincinnati"
-// )
+// archMap maps Go architecture strings to OpenShift supported values for any that differ.
+var archMap = map[string]string{
+	"amd64": "x86_64",
+}
 
 // ReleaseOptions configures either a Full or Diff mirror operation
 // on a particular release image.
 type ReleaseOptions struct {
 	cli.RootOptions
 	release string
+	arch    string
 }
 
 // NewReleaseOptions defaults ReleaseOptions.
-func NewReleaseOptions(ro cli.RootOptions) *ReleaseOptions {
-	return &ReleaseOptions{RootOptions: ro}
-}
+func NewReleaseOptions(ro cli.RootOptions, flags *pflag.FlagSet) *ReleaseOptions {
+	var arch string
+	opts := ro.FilterOptions
+	opts.Complete(flags)
+	if opts.IsWildcardFilter() {
+		arch = runtime.GOARCH
+	} else {
+		arch = strings.Split(opts.FilterByOS, "/")[1]
```

Comment:
I think users can pass something like `linux/arm/*` which would break this parser. In this case I would also either query for available platforms or match against a hard-coded list.